security/src/main/java/org/apache/geode/examples/security/Example.java

```diff
@@ -0,0 +1,156 @@
+    copyDataExample("dataAdmin", false);
+
+    printCurrentData();
+  }
+
+  /** Prints the data currently present in both /region1 and /region2 */
+  private static void printCurrentData() {
+    Example example = new Example("super-user", "123");
+    Region<String, String> region1 =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION1);
+    logger.info("======\nContents of /region1:");
+    example.getData(region1, false);
+    Region<String, String> region2 =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION2);
+    logger.info("======\nContents of /region2:");
+    example.getData(region2, false);
+
+    example.cache.close(true);
+  }
+
+
+  /**
+   * Attempt to fetch and print the data in /region1 using the authorization of the
+   * provided @username.
+   */
+  private static void getDataExample(String username, boolean expectAuthFailure) {
+    logger.info(String.format("Starting get example with user '%s', expecting %s", username,
+        expectAuthFailure ? "authorization failure." : "success."));
+
+    Example example = new Example(username, "123");
+
+    Region<String, String> region =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION1);
+
+    example.getData(region, expectAuthFailure);
+
+    example.cache.close(true);
+  }
+
+  /** Attempt to print the data present in the given @region. */
+  private void getData(Region<String, String> region, boolean expectAuthFailure) {
+    try {
+      Arrays.stream(AUTHORS).forEach(author -> logger.info(author + ": " + region.get(author)));
+    } catch (ServerOperationException e) {
+      if (e.getCause() instanceof NotAuthorizedException && expectAuthFailure) {
+        logger.info("Got ServerOperationExpectation, caused by expected NotAuthorizedException");
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Attempt to copy data from /region1 to /region2 using the authorization of the
+   * provided @username.
+   */
+  private static void copyDataExample(String username, boolean expectAuthFailure) {
+    logger.info(String.format("Starting copy example with user '%s', expecting %s", username,
+        expectAuthFailure ? "authorization failure." : "success."));
+    Example example = new Example(username, "123");
+
+    Region<String, String> region1 =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION1);
+
+    Region<String, String> region2 =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION2);
+
+    try {
+      Arrays.stream(AUTHORS).forEach(author -> region2.put(author, region1.get(author)));
+    } catch (ServerOperationException e) {
```

Comment:
Unrelated to your implementation of this example, but I wonder if the correct behavior for Geode would be to throw `NotAuthorizedException` on this `region.put()` rather than throwing a `ServerOperationException` which has a `NotAuthorizedException` as its cause.  That seems like a really unpleasant way to have to catch an auth exception.